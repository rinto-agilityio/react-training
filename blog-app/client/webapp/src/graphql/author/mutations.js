import gql from 'graphql-tag';

export const CREATE_USER = gql`
  mutation signUp($id: ID!, $email: String!, $password: String!, $name: String) {
    signUp(id: $id, email: $email, password: $password, name: $name) {
      success
      message
      author {
        id
        name
        password
        email
      }
    }
  }
`;
