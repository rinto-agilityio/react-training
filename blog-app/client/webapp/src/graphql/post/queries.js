import gql from 'graphql-tag'

export const GET_POST = gql`
  query getPostsByAuthor($authorId: ID!, $after: String, $first: Int ) {
    getPostsByAuthor(authorId: $authorId, after: $after, first: $first) {
      success
      message
      posts {
        id
        title
        content
        author {
          name
        }
      }
      pageInfo {
        endCursor
        hasNextPage
      }
      totalCount
    }
  }
`;
