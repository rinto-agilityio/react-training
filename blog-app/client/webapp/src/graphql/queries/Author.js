import gql from 'graphql-tag'

const SIGN_IN = gql`
  query signIn($email: String!, $password: String!) {
    signIn(email: $email, password: $password) {
      success
      message
      author {
        id
        name
        email
        password
        avatar
      }
    }
  }
`;

export default SIGN_IN
