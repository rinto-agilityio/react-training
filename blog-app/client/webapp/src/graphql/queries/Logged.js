import gql from 'graphql-tag'

const LOGGED_USER = gql`
  query loggedUser {
    loggedUser @client {
      email
      password
      name
    }
  }
`;

export default LOGGED_USER
