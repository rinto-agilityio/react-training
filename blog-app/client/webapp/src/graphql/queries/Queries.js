import gql from 'graphql-tag'

export const GET_AUTHORS = gql`
  query getAuthors {
		authors {
			name
			email
			id
			password
		}
  }
`;

export const LOGGED_USER = gql`
  query loggedUser {
    loggedUser @client {
      email
      password
      name
      avatar
    }
  }
`;

export const SIGN_IN = gql`
  query signIn($email: String!, $password: String!) {
    signIn(email: $email, password: $password) {
      success
      message
      author {
        id
        name
        email
        password
        avatar
      }
    }
  }
`;
