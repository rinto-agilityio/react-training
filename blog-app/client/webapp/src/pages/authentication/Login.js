import React, { useRef } from 'react';
import { ApolloConsumer } from 'react-apollo';
import { Form, Button } from 'react-bootstrap';
import { Link } from 'react-router-dom';

//component
import SignUp from './SignUp'

//query
import SIGN_IN from '../../graphql/queries/Author';

//import css
import './LoginStyle.css'

const Login = props  => {
  const email = useRef('')
  const password = useRef('')

  const handleSignIn = (event, client) => {

    event.preventDefault();

    client.query({
      query: SIGN_IN,
      variables: {
        email: email.current ? email.current.value : '',
        password: password.current ? password.current.value : ''
      }
    }).then(response => {

      const { author } = response.data.signIn

      const loginUser = {
        __typename: 'loggedUser',
        email: author.email,
        name: author.name,
        avatar: author.avatar,
        password: author.password
      };

      client.writeData({
        data: {
          loggedUser: loginUser,
        }
      });

      localStorage.setItem('userLoged', JSON.stringify(author));

      props.history.push('/');

    })
  };

  return (
    <ApolloConsumer>
      {
        client => (

          <div className="login-wrap">
            <div className="titleWrap">
              <p>Welcome To Blog</p>
            </div>
            <div className='authentication-form'>
              <Form onSubmit={(event) => handleSignIn(event, client)}>
                <Form.Group >
                  <Form.Control type='email' ref={email} placeholder='Enter email' />
                </Form.Group>
                <Form.Group >
                  <Form.Control type='password' ref={password} placeholder='Password' />
                </Form.Group>
                <Button type='submit'>Login</Button>
                <Link></Link>
              </Form>
              <p>
                Not Register?
                <Link to={'/signUp'} component={SignUp}>Create Account</Link>
              </p>
            </div>
          </div>
        )
      }
    </ApolloConsumer>
  )
}

export default Login;
