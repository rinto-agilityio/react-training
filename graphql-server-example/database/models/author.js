'use strict';
module.exports = (sequelize, DataTypes) => {
  const Author = sequelize.define('Author', {
    name: {
      type: DataTypes.STRING,
      allowNull: false
    },
    description: DataTypes.STRING,
    avatar: DataTypes.STRING
  }, {});

  Author.associate = function(models) {
    // associations can be defined here
    // https://stackoverflow.com/a/55323664/5456204
    Author.hasMany(models.Post)
  };

  return Author;
};