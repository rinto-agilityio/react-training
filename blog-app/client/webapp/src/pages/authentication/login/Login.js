import React, { useRef, useState } from 'react';
import { ApolloConsumer } from 'react-apollo';
import { Form, Button } from 'react-bootstrap';
import { Link } from 'react-router-dom';

//component
import SignUp from '../signUp/SignUp';
import ErrorMessage from '../../../components/commons/errorMessage/ErrorMessage';

//query
import { SIGN_IN } from '../../../graphql/author/queries';

//import css
import './LoginStyle.css';

const Login = props  => {
  const email = useRef('');
  const password = useRef('');
  const [errorMes, setErrorMes] = useState('');

  const handleSignIn = async (event, client) => {

    event.preventDefault();

    await client.query({
      query: SIGN_IN,
      variables: {
        email: email.current ? email.current.value : '',
        password: password.current ? password.current.value : ''
      },
      options: { fetchPolicy: 'cache-and-network' }
    }).then(response => {

      const { author } = response.data.signIn;

      client.writeData({
        data: {
          loggedUser: author,
        }
      });

      localStorage.setItem('userLoged', JSON.stringify(author));

      props.history.push('/');

    },
    (error => {
      setErrorMes(error && error.graphQLErrors && error.graphQLErrors[0].message )
    })
    );
  };
  return (
    <ApolloConsumer>
      {
        client => (

          <div className="login-wrap">
            <div className="titleWrap">
              <p>Welcome To Blog</p>
            </div>
            <div className='authentication-form'>
              <Form onSubmit={(event) => handleSignIn(event, client)}>
                <Form.Group >
                  <Form.Control type='email' ref={email} placeholder='Enter email' />
                </Form.Group>
                <Form.Group >
                  <Form.Control type='password' ref={password} placeholder='Password' />
                </Form.Group>
                {
                  errorMes &&
                  <ErrorMessage message={errorMes}/>
                }
                <Button type='submit'>Login</Button>
              </Form>
              <p>
                Not Register?
                <Link to={'/signUp'} component={SignUp}>Create Account</Link>
              </p>
            </div>
          </div>
        )
      }
    </ApolloConsumer>
  );
};

export default Login;
