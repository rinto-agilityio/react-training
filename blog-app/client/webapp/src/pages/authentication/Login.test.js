import Login from './Login';
import { MockedProvider } from 'react-apollo/test-utils';
import wait from 'waait';
import { BrowserRouter} from 'react-router-dom';
import { Form } from 'react-bootstrap';
import { SIGN_IN } from '../../graphql/author/queries';

describe('Login Component', () => {
  const props = {
    history: {
      push: jest.fn()
    }
  };

  const mocks = [
    {
      request: {
        query: SIGN_IN,
        variables: {
          email: 'rin.to@gmail.com',
          password: '123456'
        }
      },
      result: {
        data: {
          signIn: {
            success: true,
            message: 'login success',
            author: {
              id: '1557377599679',
              name: 'rinto',
              email: 'rin.to@gmail.com',
              password: '123456',
              avatar: ''
            }
          }
        }
      }
    }
  ];

  it('Test Login snapshot', () => {
    const renderComponent = renderer
      .create(
        <BrowserRouter>
          <MockedProvider mocks={mocks}>
            <Login />
          </MockedProvider>
        </BrowserRouter>
      )
      .toJSON();
    expect(renderComponent).toMatchSnapshot();
  });

  it('Test Login button should be clicked', async () => {
    const localStorageMock = {
      getItem: jest.fn(),
      setItem: jest.fn(),
      clear: jest.fn()
    };

    const component = mount(
      <BrowserRouter>
        <MockedProvider>
          <Login {...props} />
        </MockedProvider>
      </BrowserRouter>
    );
    component.find(Form).simulate('submit', {
      preventDefault: jest.fn()
    });
    await wait(10);
    expect(localStorageMock.setItem).toHaveBeenCalled();
  });

});
