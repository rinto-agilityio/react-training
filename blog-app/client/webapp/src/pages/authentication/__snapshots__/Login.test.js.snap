// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component Test Login snapshot 1`] = `
<div
  className="login-wrap"
>
  <div
    className="titleWrap"
  >
    <p>
      Welcome To Blog
    </p>
  </div>
  <div
    className="authentication-form"
  >
    <form
      className=""
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <input
          className="form-control"
          placeholder="Enter email"
          type="email"
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="form-control"
          placeholder="Password"
          type="password"
        />
      </div>
      
      <button
        className="btn btn-primary"
        disabled={false}
        type="submit"
      >
        Login
      </button>
    </form>
    <p>
      Not Register?
      <a
        component={[Function]}
        href="/signUp"
        onClick={[Function]}
      >
        Create Account
      </a>
    </p>
  </div>
</div>
`;
